### PR TITLE
Move response signing from `processTransactions` handler to `callApi`

### PR DESCRIPTION
**Pull request by amarthadan**
Issue [AN-397](https://api3dao.atlassian.net/browse/AN-397)

### CODE REVIEW

**Review by Siegrift — APPROVED** [review]
LGTM :+1:  Doing it this way also feels a bit more natural to me.

**Review by andreogle — APPROVED** [review]
Nice one 👍🏻

File: packages/airnode-node/src/evm/fulfillments/api-calls.ts [hunk 1]
```diff
@@ -123,12 +111,8 @@ async function testAndSubmitFulfill(
     return [submitLogs, submitErr, submitData];
   }
 
-  // Compute signature once.
-  // Not computing this beforehand because it might be a very computational expensive operation
-  const signature = await signResponseMessage(request, options.masterHDNode);
-
   // Should not throw
-  const [testLogs, testErr, testData] = await testFulfill(airnodeRrp, { ...request, signature }, options);
+  const [testLogs, testErr, testData] = await testFulfill(airnodeRrp, { ...request }, options);
```
Comment (acenolaza) [inline packages/airnode-node/src/evm/fulfillments/api-calls.ts:115]: Late but I was wondering if we really need to spread the `request` object here or if we could just pass it as is to `testFulfill()`

File: packages/airnode-node/src/evm/fulfillments/api-calls.ts [hunk 2]
```diff
@@ -149,7 +133,7 @@ async function testAndSubmitFulfill(
 
   // We expect the transaction to be successful if submitted
   if (testData?.callSuccess) {
-    const [submitLogs, submitErr, submitData] = await submitFulfill(airnodeRrp, { ...request, signature }, options);
+    const [submitLogs, submitErr, submitData] = await submitFulfill(airnodeRrp, { ...request }, options);
```
Comment (acenolaza) [inline packages/airnode-node/src/evm/fulfillments/api-calls.ts:136]: same here

File: packages/airnode-node/src/evm/fulfillments/api-calls.ts (hunk 1 above)
Comment (amarthadan) [reply packages/airnode-node/src/evm/fulfillments/api-calls.ts:115]: Of course, we can :man_facepalming: I'll add it to my next PR.

**Pull request merged**